app/src/main/java/com/nightscout/android/upload/Uploader.java

```diff
@@ -160,15 +161,31 @@ private boolean upload(List<GlucoseDataSet> glucoseDataSets,
         for (BaseUploader uploader : uploaders) {
             // TODO(klee): capture any exceptions here so that all configured uploaders will attempt
             // to upload
-            allSuccessful &= uploader.uploadRecords(glucoseDataSets, meterRecords, calRecords, deviceStatus);
-            reporter.report(EventType.UPLOADER, EventSeverity.INFO,
-                    String.format(context.getString(R.string.event_success_upload),
-                            uploader.getIdentifier()));
-
+            try {
+                allSuccessful &= uploader.uploadRecords(glucoseDataSets, meterRecords, calRecords, deviceStatus);
+                reporter.report(EventType.UPLOADER, EventSeverity.INFO,
+                        String.format(context.getString(R.string.event_success_upload),
+                                uploader.getIdentifier()));
+            } catch (MongoException e) {
+                if (e.getCode() == 18) {
```

Comment:
Could you also do reviews for the guys who wrote the mongo driver please? :wink: